File: lingting-core/src/main/java/live/lingting/framework/crypto/mac/MacBuilder.java
package live.lingting.framework.crypto.mac;

/**
 * @author lingting 2024-09-04 13:42
 */
public class MacBuilder extends AbstractMacBuilder<MacBuilder> {

	public static class Hmac extends AbstractMacBuilder<Hmac> {

		public Hmac sha256() {
			return algorithm("HmacSHA256");
		}

		public Hmac sha1() {
			return algorithm("HmacSHA1");
		}

	}

}

File: lingting-core/src/main/kotlin/live/lingting/framework/crypto/mac/MacBuilder.kt
package live.lingting.framework.crypto.mac

/**
 * @author lingting 2024-09-04 13:42
 */
class MacBuilder : AbstractMacBuilder<MacBuilder>() {
    class Hmac : AbstractMacBuilder<Hmac>() {
        fun sha256(): Hmac {
            return algorithm("HmacSHA256")
        }

        fun sha1(): Hmac {
            return algorithm("HmacSHA1")
        }
    }
}
